### PR TITLE
docs: refactor uml packages

### PR DESCRIPTION
**Pull request by joseluis-teran-jalafund**
## Trello Ticket

**[ID-22](https://trello.com/c/wtLOO6EG/31-id-22-update-uml-layers)**

## What did I do?

Update UML packages between layers, so we have a better infrastructure in the project.

## Why did I do this?

In order to present the layered architecture we have in **develop** branch.

## How did I do it?

Divide the `.puml` file into new ones in order to represent each one a different layer.

## (Optional) Notes:

Comments that should be taken into account during code reviews regarding the pull request.

### CODE REVIEW

File: docs/uml/data-access.puml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+@startuml Data Access
+
+package "Data Access Layer" {
+    interface IBaseRepository {
+        + GetAll(T entity) : Task<int>
+        + GetById(T entity) : Task<int>
+        + Create(T entity) : Task<int>
+        + Update(T entity) : Task<int>
+        + Delele(T entity) : Task<bool>
```
Comment (gabriela-garcia-jalafund) [inline docs/uml/data-access.puml]: in CRUD operations:

      + GetAll(T entity) : Task<int>
        + GetById(T entity) : Task<int>
        + Create(T entity) : Task<int>
        + Update(T entity) : Task<int>
        + Delele(T entity) : Task<bool>



It is better to keep in mind that when we ask for a list it is not an int, it is a list, we can use IEnumerable or IList, for getById, we ask for an id, not an entity, the same for delete

 + GetAll() : Task<IEnumerable<T>>  
        + GetById(T id) : Task<T>
        + Create(T entity) : Task<int>
        + Update(T entity) : Task<int>
        + Delele(T id) : Task<bool>

**Review by gabriela-garcia-jalafund — APPROVED** [review]

**Review by alex-paca-jalafund — COMMENTED** [review]
Its there a reason why the Goal and Budget class has no relation with other classes?

**Comment by gabriela-garcia-jalafund** [comment]
Is there a reason why the Goal and Budget class has no relation with other classes? @alex-paca-jalafund 
It is because at the moment it is not defined, we prioritize the two objectives that we have: Phase 1 - Setup & User Authentication:
Phase 2 - Basic Expense and Income CRUD
Also, due to the time issue, if we have time, then we could modify it without any problem.

**Review by gabriela-garcia-jalafund — APPROVED** [review]

**Review by alex-paca-jalafund — APPROVED** [review]
Nice work.

**Pull request merged**